creator/nx_creator_ptycho.py

```diff
@@ -265,16 +265,14 @@ def create_transformation_group(self, h5parent: h5py.Group):
     def create_axis(
         self,
         transformation: h5py.Group,
-        axis_name: str,
-        value: np.ndarray,
+        value: str,
```

Comment:
From the example and NXtransformation definition, I believe we would create an anonymous dataset for each axis?

Reply:
I'm not sure what you mean by anonymous. Each axis must have a name because of the `depends_on` attributes use this name to create a chain of transformations. The value of each axis must be an number (scalar or vector).

Reply:
Please look closely at the example I created in 'test.py'

Reply:
Yes, it's clear that the axis needs a name. However, what should the value be then? The example here: https://manual.nexusformat.org/design.html#coordinate-transformation-field-and-attributes uses the value itself to set the name so the axis_name key-value pair is not being used:
```
sample:NXsample
    transforms:NXtransformations
      rotation_angle
        @transformation_type=rotation
        @vector=0,1,0
        @offset=0,0,0
        @depends_on=.
      chi
        @transformation_type=rotation
        @vector=0,0,1
        @offset=0,0,0
        @depends_on=rotation_angle
      phi
        @transformation_type=rotation
        @vector=0,1,0
        @offset=0,0,0
        @depends_on=chi
    depends_on 
      transforms/phi
```
In your example you use values of certain axis as the value for the value entry in the Geometry group which would then be redundant to the offset values? That's probably one way to do it, I don't see this intended use in the example/definition though.

Reply:
If you look at [the class description of the NXtransformation](https://manual.nexusformat.org/classes/base_classes/NXtransformations.html#nxtransformations). It's much clearer that each axis is a dataset whose value is position(s) along the axis.

The offset is not redundant. The offset is a single 3D vector. Each axis value is a 1D array of positions along the axis.

The example should actually be as follows:
```
sample:NXsample
    transforms:NXtransformations
      rotation_angle:NXnumber
        @transformation_type=rotation
        @vector=0,1,0
        @offset=0,0,0
        @depends_on=.
      chi:NXnumber
        @transformation_type=rotation
        @vector=0,0,1
        @offset=0,0,0
        @depends_on=rotation_angle
      phi:NXnumber
        @transformation_type=rotation
        @vector=0,1,0
        @offset=0,0,0
        @depends_on=chi
    depends_on 
      transforms/phi
```